contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5Utils.java

```diff
@@ -0,0 +1,248 @@
+      //Fall back to string
+      if(name.equalsIgnoreCase("OTHER")) {
+        return MinorType.GENERIC_OBJECT;
+      } else {
+        return MinorType.VARCHAR;
+      }
+    } else if (type.isAssignableFrom(long.class)) {
+      return MinorType.BIGINT;
+    } else if (type.isAssignableFrom(short.class)) {
+      return MinorType.SMALLINT;
+    } else if (type.isAssignableFrom(byte.class)) {
+      return MinorType.TINYINT;
+    } else if (type.isAssignableFrom(int.class)) {
+      return MinorType.INT;
+    } else if (type.isAssignableFrom(double.class) || type.isAssignableFrom(float.class)) {
+      return MinorType.FLOAT8;
+    } else if (type.isAssignableFrom(String.class)) {
+      return MinorType.VARCHAR;
+    } else if (type.isAssignableFrom(java.util.Date.class)) {
+      return MinorType.TIMESTAMP;
+    } else if (type.isAssignableFrom(boolean.class) || type.isAssignableFrom(BitSet.class)) {
+      return MinorType.BIT;
+    } else if (type.isAssignableFrom(Map.class)) {
+      return MinorType.MAP;
+    } else if (type.isAssignableFrom(Enum.class)) {
+      return MinorType.GENERIC_OBJECT;
+    }
+    return MinorType.GENERIC_OBJECT;
+  }
+
+  protected static Class<?> getDatasetClass(String path, IHDF5Reader reader) {
+    HDF5DataSetInformation info = reader.getDataSetInformation(resolvePath(path, reader));
+    Class<?> dataClass = info.getTypeInformation().tryGetJavaType();
+    return dataClass;
+  }
+
+  protected static String resolvePath(String path, IHDF5Reader reader) {
+    if (reader.exists(path)) {
+      // Resolve references, if any.
+      if (reader.isDataSet(path)) {
+        HDF5DataClass dataClass = reader.getDataSetInformation(path).getTypeInformation().getDataClass();
+        if (dataClass.toString().equals("REFERENCE")) {
+          return reader.readObjectReference(path);
+        }
+      }
+      return path;
+    } else if (!caseSensitive) {
+      // Look for a match with a different case.
+      String[] pathParts = path.split("/");
+      String resolvedPath = "/";
+      for (int i = 1; i < pathParts.length; i++) {
+        String testPath = (resolvedPath.endsWith("/")) ? resolvedPath + pathParts[i] : resolvedPath + "/" + pathParts[i];
+        if (reader.exists(testPath)) {
+          resolvedPath = testPath;
+        } else if (reader.isGroup(resolvedPath)) {
+          List<String> children = reader.getGroupMembers(resolvedPath);
+          for (String child : children) {
+            if (child.equalsIgnoreCase(pathParts[i])) {
+              resolvedPath = resolvedPath + "/" + child;
+            }
+          }
+        }
+      }
+      if (path.equalsIgnoreCase(resolvedPath)) {
+        return resolvedPath;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * This helper function returns the name of a HDF5 record from a data path
+   *
+   * @param path Path to HDF5 data
+   * @return String name of data
+   */
+  public static String getNameFromPath(String path) {
```

Comment:
I think you already have similar method, extract to util class to use on both places

Reply:
Fixed